addons/fleet_account/models/fleet_vehicle.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.exceptions import AccessError
+
+
+class FleetVehicle(models.Model):
+    _inherit = 'fleet.vehicle'
+
+    analytic_account_id = fields.Many2one('account.analytic.account', string='Analytic Account', copy=False, ondelete='set null',
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", check_company=True,
```

Comment:
I think that check_company actually adds your domain. See: https://github.com/odoo/odoo/blame/master/odoo/fields.py#L2332